src/components/DateInput/dateInput.test.js

```diff
@@ -54,88 +56,84 @@ describe('DateInput component', () => {
   describe('input and picker interation', () => {
     let dateChangeFn;
     let inputChangeFn;
-    let wrapper;
-    let input;
 
     beforeEach(() => {
       dateChangeFn = jest.fn();
       inputChangeFn = jest.fn();
-      wrapper = mount(<DateInput onDateChange={dateChangeFn} onInputChange={inputChangeFn} />);
-      wrapper.find('input').simulate('click'); // open the picker for every test
-      input = wrapper.find('input');
+      render(<DateInput onDateChange={dateChangeFn} onInputChange={inputChangeFn} />);
+      fireEvent.click(screen.getByLabelText('Date Input'));
     });
 
     afterEach(() => {
       dateChangeFn = null;
       inputChangeFn = null;
-      wrapper = null;
-      input = null;
     });
 
-    it('calls onInputChange when the input value changes', () => {
+    it('calls onInputChange when the input value changes', async () => {
       expect(inputChangeFn).not.toHaveBeenCalled();
-      input.simulate('change', { target: { value: '4' } });
+      await userEvent.type(screen.getByRole('textbox'), '4')
       expect(inputChangeFn).toHaveBeenCalled();
     });
 
-    it('calls onDateChange when user types a VALID freeform date', () => {
+    it('calls onDateChange when user types a VALID freeform date', async () => {
       expect(dateChangeFn).not.toHaveBeenCalled();
-      input.simulate('change', { target: { value: '4/20/2020' } });
+      await userEvent.type(screen.getByRole('textbox'), '4/20/2020')
       expect(dateChangeFn).toHaveBeenCalled();
     });
 
-    it('does not call onDateChange when user types an INCOMPLETE freeform date', () => {
+    it('does not call onDateChange when user types an INCOMPLETE freeform date', async () => {
       expect(dateChangeFn).not.toHaveBeenCalled();
-      input.simulate('change', { target: { value: '2/3' } });
+      await userEvent.type(screen.getByRole('textbox'), '2/3')
       expect(dateChangeFn).not.toHaveBeenCalled();
     });
 
-    it('does not call onDateChange when user types an INVALID freeform date', () => {
+    it('does not call onDateChange when user types an INVALID freeform date', async () => {
       expect(dateChangeFn).not.toHaveBeenCalled();
-      input.simulate('change', { target: { value: '2/30/2020' } });
+      await userEvent.type(screen.getByRole('textbox'), '2/30/2020')
       expect(dateChangeFn).not.toHaveBeenCalled();
     });
 
     it('calls onDateChange when user selects a day in the picker', () => {
       expect(dateChangeFn).not.toHaveBeenCalled();
-      wrapper.find('Day').at(7).simulate('click');
+      fireEvent.click(screen.getByText('7'));
```

Comment:
This is me being lazy. Technically `screen.getByText` should be your absolute last resort, and almost should never be used to grab something you then want to subsequently click on. I'm probably going to merge "as is" so that this comment / example can stay here for posterity & reference.

Reply:
Is this done because there's no clear aria role to use for selection?

Reply:
this is sort of silly but to give you the short answer: it shouldn't be done like this and there's probably an accessible hook. I figured it was instructive to show me doing it the wrong way

To give the full answer (only read if you're bored or are really interested):

So when I was trying to test this, `screen.debug()` wasn't outputting the full source. So I sort of "guessed" that the text of 7 would be in the list, and the test passed. Later on when I figured out how to get `screen.debug()` to output the full source, I realized the PR could be used as a tutorial of sorts. Given that, I sort of wanted to leave this here to show people both what you shouldn't do, but also at the same time show this selector off because it's likely people will need it when doing tests on non interactive stuff. My plan is to basically fast follow this PR with a fix specifically to this line 🙃. It's silly reasoning but I figure it's well intentioned and I'm leaving the test much better than what it was.

Reply:
just seeing your response now sorry! Thanks for the update, and follow up PR!